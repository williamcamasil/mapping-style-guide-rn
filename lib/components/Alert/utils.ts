import { ViewStyle } from 'react-native';

import { Icons } from '../../assets';
import { AppThemeType } from '../../theme/default';
import { ColorPaletesType } from '../../tokens';

export type AlertVariantType =
  'success' |
  'info' |
  'warning' |
  'error';

export const getVariantColorStyle = (variant?: AlertVariantType): ColorPaletesType => {
  switch (variant) {
    case 'warning':
      return 'feedbackAlert500';
    case 'error':
      return 'feedbackError500';
    case 'info':
      return 'neutralGray600';
    case 'success':
    default:
      return 'feedbackSuccess500';
  }
};

export const getDefaultIcon = (variant?: AlertVariantType) => {
  switch (variant) {
    case 'success':
      return Icons.Default.Success;
    case 'info':
      return Icons.Default.Information;
    case 'warning':
    case 'error':
    default:
      return Icons.Default.Warning;
  }
};

export const getAlertStyle = (
  theme: AppThemeType,
  variant?: AlertVariantType,
): ViewStyle => {

  switch (variant) {
    case 'warning':
      return {
        backgroundColor: theme.colors.feedbackAlert100,
      };
    case 'error':
      return {
        backgroundColor: theme.colors.feedbackError100,
      };
    case 'info':
      return {
        backgroundColor: theme.colors.neutralGray100,
      };
    case 'success':
    default:
      return {
        backgroundColor: theme.colors.feedbackSuccess100,
      };
  }
};
