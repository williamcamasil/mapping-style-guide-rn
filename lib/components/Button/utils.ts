import { ViewStyle } from 'react-native';

import { AppThemeType } from '../../theme/default';
import { ColorPaletesType } from '../../tokens';

export type ButtonVariantType =
  'containedPrimary' |
  'containedSecondary' |
  'outlinedPrimary' |
  'outlinedSecondary' |
  'text';

export type ButtonSizeType =
  'large' |
  'medium' |
  'small';

export const getVariantStyle = (
  theme: AppThemeType,
  variant?: ButtonVariantType,
  disabled?: boolean | null,
): ViewStyle => {
  switch (variant) {
    case 'containedSecondary':
      return {
        backgroundColor: disabled ? theme.colors.primary100 : theme.colors.primary200,
        borderWidth: theme.borders.width.none,
      };
    case 'outlinedPrimary':
      return {
        backgroundColor: 'transparent',
        borderWidth: theme.borders.width.thin,
        borderColor: disabled ? theme.colors.primary100 : theme.colors.primary600,
      };
    case 'outlinedSecondary':
      return {
        backgroundColor: 'transparent',
        borderWidth: theme.borders.width.thin,
        borderColor: disabled ? theme.colors.neutralGray100 : theme.colors.neutralGray300,
      };
    case 'text':
      return {
        backgroundColor: disabled ? theme.colors.neutralGray100 : 'transparent',
        borderWidth: theme.borders.width.none,
      };
    case 'containedPrimary':
    default:
      return {
        backgroundColor: disabled ? theme.colors.primary100 : theme.colors.primaryMain,
        borderWidth: theme.borders.width.none,
      };
  }
};

export const getTextColor = (
  variant?: ButtonVariantType,
  disabled?: boolean | null,
): ColorPaletesType => {
  switch (variant) {
    case 'containedSecondary':
      return disabled ? 'primary300' : 'primaryMain';
    case 'outlinedPrimary':
      return disabled ? 'primary300' : 'primaryMain';
    case 'outlinedSecondary':
      return disabled ? 'neutralGray200' : 'neutralGray600';
    case 'text':
      return disabled ? 'neutralGray400' : 'neutralGray500';
    case 'containedPrimary':
    default:
      return disabled ? 'primary300' : 'neutralWhite';
  }
};

export const getSizeStyle = (theme: AppThemeType, size?: ButtonSizeType, hasText?: boolean): ViewStyle => {
  switch (size) {
    case 'large':
      return {
        height: 64,
        minWidth: 64,
        paddingHorizontal: hasText ? theme.spacings.sXL : undefined,
      };
    case 'small':
      return {
        height: 30,
        minWidth: 30,
        paddingHorizontal: hasText ? theme.spacings.sXS : undefined,
      };
    case 'medium':
    default:
      return {
        height: 48,
        minWidth: 48,
        paddingHorizontal: hasText ? theme.spacings.sMedium : undefined,
      };
  }
};

export const getStylesToBlock = (block: boolean | undefined): ViewStyle | null => {
  if (block) return null;

  return {
    flexDirection: 'row',
  };
};
