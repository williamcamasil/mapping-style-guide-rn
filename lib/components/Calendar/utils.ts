import { dateToString } from 'mapping-context-rn';
import type { MarkedDates, Theme } from 'react-native-calendars/src/types';

import { AppThemeType } from '../../theme';

type ExtendedTheme = Theme & {
  'stylesheet.calendar.header': object;
  'stylesheet.day.basic': object;
};

/**
 * Define o tipo de data para o formato yyyy-MM-dd
 */
export type StringDateType = `${number}${number}${number}${number}-${number}${number}-${number}${number}`;

export const DATE_FORMAT = 'yyyy-MM-dd';

export const DAY_SIZE = 42;

export const getDefaultValue = (value?: StringDateType | null) => (
  value ?? dateToString(new Date(), DATE_FORMAT) as StringDateType
);

export const getDisabledDates = (disabledDates?: StringDateType[]) => {
  if (!disabledDates?.length) return {};
  const dates: MarkedDates = {};
  disabledDates.forEach((date: StringDateType) => {
    dates[date] = {
      disabled: true,
      disableTouchEvent: true,
    };
  });
  return dates;
};

export const getCommonTheme = (theme: AppThemeType): ExtendedTheme => ({
  selectedDayBackgroundColor: theme.colors.primaryMain,
  todayTextColor: theme.colors.primaryMain,
  dayTextColor: theme.colors.neutralGray600,
  textDayFontFamily: theme.typography.weights.bold.fontFamily,
  textMonthFontFamily: theme.typography.weights.bold.fontFamily,
  textDayHeaderFontFamily: theme.typography.weights.bold.fontFamily,
  textDayFontWeight: theme.typography.weights.bold.fontWeight,
  textMonthFontWeight: theme.typography.weights.bold.fontWeight,
  textDayHeaderFontWeight: theme.typography.weights.bold.fontWeight,
  textDayFontSize: theme.typography.sizes.small,
  textMonthFontSize: theme.typography.sizes.small,
  textDayHeaderFontSize: theme.typography.sizes.small,
  'stylesheet.calendar.header': {
    dayHeader: {
      marginTop: theme.spacings.sXS,
      marginBottom: theme.spacings.sXS,
      textAlign: 'center',
      fontSize: theme.typography.sizes.small,
      color: theme.colors.primaryMain,
      fontFamily: theme.typography.weights.bold.fontFamily,
      fontWeight: theme.typography.weights.bold.fontWeight,
      flex: 1,
    },
    week: {
      flexDirection: 'row',
      justifyContent: 'space-between',
    },
  },
  'stylesheet.day.basic': {
    base: {
      height: DAY_SIZE,
      width: DAY_SIZE,
      borderBottomLeftRadius: theme.borders.radius.XL,
      borderBottomRightRadius: theme.borders.radius.XL,
      borderTopLeftRadius: theme.borders.radius.XL,
      borderTopRightRadius: theme.borders.radius.XL,
      alignItems: 'center',
      justifyContent: 'center',
    },
  },
});
