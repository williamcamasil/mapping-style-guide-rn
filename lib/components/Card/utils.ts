import { ColorValue, ViewStyle } from 'react-native';

import { AppThemeType } from '../../theme/default';

export type CardSizeType =
  'small' |
  'big';

export type CardVariantType =
  'default' |
  'highlighted50' |
  'highlighted100';

export const getCardSizeStyle = (theme: AppThemeType, size?: CardSizeType): ViewStyle => {
  switch (size) {
    case 'small':
      return {
        width: 135,
        height: 120,
        padding: theme.spacings.sXS,
      };
    case 'big':
    default:
      return {
        minHeight: 128,
        padding: theme.spacings.sSmall,
      };
  }
};

export const getCardShadowStyle = (theme: AppThemeType, showShadow?: boolean): ViewStyle | null => {
  if (showShadow) {
    return theme.shadows.level4;
  }

  return {
    borderWidth: theme.borders.width.thin,
    borderColor: theme.colors.neutralGray200,
  };
};

export const getCardVariantStyle = (theme: AppThemeType, cardType?: CardVariantType): ViewStyle | null => {
  switch (cardType) {
    case 'highlighted50':
      return {
        borderWidth: theme.borders.width.thin,
        borderColor: theme.colors.primary700,
      };
    case 'highlighted100':
      return {
        borderWidth: theme.borders.width.thin,
        borderColor: theme.colors.primary600,
        backgroundColor: theme.colors.primaryMain,
      };
    case 'default':
    default:
      return null;
  }
};

export const getCardCustomBorderStyle = (theme: AppThemeType, borderColor?: ColorValue): ViewStyle | null => {
  if (!borderColor) return null;

  return {
    borderWidth: theme.borders.width.thin,
    borderColor,
  };
};

export const getArrowColor = (theme: AppThemeType, variant?: CardVariantType, arrowColor?: ColorValue): ColorValue => {
  if (arrowColor) return arrowColor;

  if (variant === 'highlighted100') return theme.colors.neutralWhite;

  return theme.colors.neutralGray500;
};
