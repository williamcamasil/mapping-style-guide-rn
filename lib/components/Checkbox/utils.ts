import { CheckBoxImgPropsType } from '../../assets/images/CheckboxImg';
import { AppThemeType } from '../../theme/default';
import { ColorPaletesType } from '../../tokens';

export type CheckboxVariantType = 'default' | 'secondary';

export const getTextColor = (error?: boolean, variant?: CheckboxVariantType, disabled?: boolean): ColorPaletesType => {
  if (disabled) {
    if (variant === 'secondary') return 'secondary500';
    return 'neutralGray300';
  }

  if (error) {
    if (variant === 'secondary') return 'feedbackError300';
    return 'feedbackError400';
  }

  if (variant === 'secondary') return 'secondary200';

  return 'neutralGray700';
};

const getBackgroundColorStyle = (theme: AppThemeType, variant?: CheckboxVariantType) => {
  if (variant === 'secondary') return theme.colors.secondary700;

  return theme.colors.neutralGray100;
};

const getCheckedBorderColorStyle = (theme: AppThemeType, variant?: CheckboxVariantType) => {
  if (variant === 'secondary') return theme.colors.secondary600;

  return theme.colors.neutralGray200;
};

const getColorStyle = (theme: AppThemeType, error?: boolean, variant?: CheckboxVariantType) => {
  if (error) {
    if (variant === 'secondary') return theme.colors.feedbackError300;
    return theme.colors.feedbackError400;
  }

  return theme.colors.primaryMain;
};

const getDisabledBorderColorStyle = (theme: AppThemeType, variant?: CheckboxVariantType) => {
  if (variant === 'secondary') return theme.colors.secondary500;

  return theme.colors.neutralGray200;
};

export const getBoxStyle = (
  theme: AppThemeType,
  error?: boolean,
  disabled?: boolean,
  checked?: boolean,
  variant?: CheckboxVariantType,
): CheckBoxImgPropsType => {
  if (disabled) {
    return (
      {
        backgroundColor: 'transparent',
        borderColor: getDisabledBorderColorStyle(theme, variant),
        borderWidth: theme.borders.width.thin,
        checkColor: theme.colors.neutralGray200,
        checked,
      }
    );
  }

  if (checked) {
    return (
      {
        backgroundColor: getColorStyle(theme, error, variant),
        borderColor: getCheckedBorderColorStyle(theme, variant),
        borderWidth: theme.borders.width.thick,
        checkColor: theme.colors.neutralWhite,
        checked,
      }
    );
  }

  return (
    {
      backgroundColor: getBackgroundColorStyle(theme, variant),
      borderColor: getColorStyle(theme, error, variant),
      borderWidth: theme.borders.width.thin,
      checkColor: theme.colors.neutralGray100,
      checked,
    }
  );
};
