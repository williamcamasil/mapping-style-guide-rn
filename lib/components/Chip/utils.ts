import { AppThemeType } from '../../theme/default';

export const getButtonStyles = (selected: boolean, theme: AppThemeType) => {
  if (selected) {
    return {
      backgroundColor: theme.colors.primary100,
      borderColor: theme.colors.primary300,
    };
  }
  return {
    backgroundColor: 'transparent',
    borderColor: theme.colors.neutralGray300,
  };
};
