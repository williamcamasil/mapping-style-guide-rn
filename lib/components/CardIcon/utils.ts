import { ColorValue } from 'react-native';

import { AppThemeType } from '../../theme/default';
import { CardVariantType } from '../Card/utils';

export const getIconColor = (
  theme: AppThemeType,
  customColor?: ColorValue,
  variant?: CardVariantType,
): ColorValue | string => {
  if (customColor) return customColor;

  if (variant === 'highlighted100') {
    return theme.colors.neutralWhite;
  }

  return theme.colors.primaryMain;
};
