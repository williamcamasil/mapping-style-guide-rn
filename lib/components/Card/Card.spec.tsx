import React from 'react';
import { ViewStyle } from 'react-native';

import { render, fireEvent } from '@testing-library/react-native';

import { Card } from '..';

import { DefaultTheme } from '../../theme';
import { getCardCustomBorderStyle, getCardSizeStyle, getCardVariantStyle } from './utils';

describe('card snapshot', () => {
  it('default', () => {
    const tree = render(
      <Card />,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('card small', () => {
    const tree = render(
      <Card size="small" />,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('card no shadow', () => {
    const tree = render(
      <Card showShadow={false} />,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('card custow arrow color', () => {
    const tree = render(
      <Card showArrow arrowColor="red" />,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('card arrow highlighted100', () => {
    const tree = render(
      <Card showArrow variant="highlighted100" />,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('card big', () => {
    const tree = render(
      <Card size="big" />,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('card disabled', () => {
    const tree = render(
      <Card disabled />,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('card type highlighted50', () => {
    const result = render(
      <Card variant="highlighted50" />,
    );

    const card = result.getByTestId('card-touchable');

    expect(card).toBeDefined();
    expect(card.props?.style.borderWidth).toEqual(DefaultTheme.borders.width.thin);
    expect(card.props?.style.borderColor).toEqual(DefaultTheme.colors.primary700);
    expect(result.toJSON()).toMatchSnapshot();
  });

  it('card type highlighted100', () => {
    const result = render(
      <Card variant="highlighted100" />,
    );

    const card = result.getByTestId('card-touchable');

    expect(card).toBeDefined();
    expect(card.props?.style.borderWidth).toEqual(DefaultTheme.borders.width.thin);
    expect(card.props?.style.borderColor).toEqual(DefaultTheme.colors.primary600);
    expect(card.props?.style.backgroundColor).toEqual(DefaultTheme.colors.primaryMain);
    expect(result.toJSON()).toMatchSnapshot();
  });

  it('card with custom style', () => {
    const result = render(
      <Card style={{ borderWidth: 1, borderColor: DefaultTheme.colors.primary700 }} />,
    );

    const card = result.getByTestId('card-touchable');

    expect(card).toBeDefined();
    expect(card.props?.style.borderWidth).toEqual(1);
    expect(card.props?.style.borderColor).toEqual(DefaultTheme.colors.primary700);
    expect(result.toJSON()).toMatchSnapshot();
  });
});

describe('Card events', () => {
  it('onPress enabled', () => {
    const handlePress = jest.fn();

    const result = render(
      <Card
        onPress={handlePress}
      >
        Card mock
      </Card>,
    );

    const touchable = result.getByTestId('card-touchable');

    fireEvent.press(touchable);

    expect(handlePress).toHaveBeenCalled();
  });

  it('onPress disabled', () => {
    const handlePress = jest.fn();

    const result = render(
      <Card
        onPress={handlePress}
        disabled
      >
        Card mock
      </Card>,
    );

    const touchable = result.getByTestId('card-touchable');

    fireEvent.press(touchable);

    expect(handlePress).not.toHaveBeenCalled();
  });

  it('show on loading', async () => {
    const handlePress = jest.fn();

    const result = render(
      <Card
        onPress={handlePress}
        loading
      >
        Card mock
      </Card>,
    );

    const touchables = result.queryByTestId('card-touchable');

    const shimmerId = result.getByTestId('shimmer-id');

    expect(shimmerId).toBeTruthy();
    expect(touchables).toBeNull();
  });
});

describe('utils', () => {
  it('getCardSizeStyle Big', () => {
    const sizeStyleMock: ViewStyle = {
      minHeight: 128,
      padding: 20,
    };

    const sizeStyle = getCardSizeStyle(DefaultTheme, 'big');

    expect(sizeStyle).toEqual(sizeStyleMock);
  });

  it('getCardSizeStyle Small', () => {
    const sizeStyleMock: ViewStyle = {
      width: 135,
      padding: 16,
      height: 120,
    };

    const sizeStyle = getCardSizeStyle(DefaultTheme, 'small');

    expect(sizeStyle).toEqual(sizeStyleMock);
  });

  it('getCardTypeStyle default', () => {
    const cardTypeStyle = getCardVariantStyle(DefaultTheme, 'default');

    expect(cardTypeStyle).toEqual(null);
  });

  it('getCardTypeStyle highlighted50', () => {
    const cardTypeStyleMock: ViewStyle = {
      borderWidth: DefaultTheme.borders.width.thin,
      borderColor: DefaultTheme.colors.primary700,
    };

    const cardTypeStyle = getCardVariantStyle(DefaultTheme, 'highlighted50');

    expect(cardTypeStyle).toEqual(cardTypeStyleMock);
  });

  it('getCardTypeStyle highlighted100', () => {
    const cardTypeStyleMock: ViewStyle = {
      borderWidth: DefaultTheme.borders.width.thin,
      borderColor: DefaultTheme.colors.primary600,
      backgroundColor: DefaultTheme.colors.primaryMain,
    };

    const cardTypeStyle = getCardVariantStyle(DefaultTheme, 'highlighted100');

    expect(cardTypeStyle).toEqual(cardTypeStyleMock);
  });

  it('getCardCustomBorderStyle with custom borderColor param', () => {
    const borderColor: string = DefaultTheme.colors.feedbackAlert400;

    const cardBorderStyleMock: ViewStyle = {
      borderWidth: DefaultTheme.borders.width.thin,
      borderColor,
    };

    const cardBorderStyle = getCardCustomBorderStyle(DefaultTheme, borderColor);

    expect(cardBorderStyle).toEqual(cardBorderStyleMock);
  });

  it('getCardCustomBorderStyle with border param undefined', () => {
    const cardBorderStyle = getCardCustomBorderStyle(DefaultTheme);

    expect(cardBorderStyle).toBeNull();
  });
});
