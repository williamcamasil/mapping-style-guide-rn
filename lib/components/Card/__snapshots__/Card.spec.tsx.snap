// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`card snapshot card arrow highlighted100 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "backgroundColor": "#013188",
      "borderColor": "#2453A9",
      "borderRadius": 12,
      "borderWidth": 1,
      "elevation": 7,
      "flexDirection": "row",
      "minHeight": 128,
      "opacity": 1,
      "padding": 20,
      "shadowColor": "#000",
      "shadowOffset": {
        "height": 3,
        "width": 0,
      },
      "shadowOpacity": 0.29,
      "shadowRadius": 4.65,
    }
  }
  testID="card-touchable"
>
  <View
    style={
      {
        "flex": 1,
      }
    }
  />
  <View
    style={
      {
        "bottom": 0,
        "justifyContent": "center",
        "position": "absolute",
        "right": 20,
        "top": 0,
      }
    }
  >
    <RNSVGSvgView
      align="xMidYMid"
      bbHeight="16"
      bbWidth="16"
      fill="none"
      focusable={false}
      height={16}
      meetOrSlice={0}
      minX={0}
      minY={0}
      style={
        [
          {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          {
            "flex": 0,
            "height": 16,
            "width": 16,
          },
        ]
      }
      vbHeight={16}
      vbWidth={16}
      width={16}
    >
      <RNSVGGroup
        fill={null}
        propList={
          [
            "fill",
          ]
        }
      >
        <RNSVGPath
          d="m10 12 4-4-4-4"
          fill={
            {
              "payload": 4278190080,
              "type": 0,
            }
          }
          propList={
            [
              "stroke",
              "strokeWidth",
              "strokeLinecap",
              "strokeLinejoin",
            ]
          }
          stroke={
            {
              "payload": 4294967295,
              "type": 0,
            }
          }
          strokeLinecap={1}
          strokeLinejoin={1}
          strokeWidth="2"
        />
      </RNSVGGroup>
    </RNSVGSvgView>
  </View>
</View>
`;

exports[`card snapshot card big 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "backgroundColor": "#FFFFFF",
      "borderRadius": 12,
      "elevation": 7,
      "flexDirection": "row",
      "minHeight": 128,
      "opacity": 1,
      "padding": 20,
      "shadowColor": "#000",
      "shadowOffset": {
        "height": 3,
        "width": 0,
      },
      "shadowOpacity": 0.29,
      "shadowRadius": 4.65,
    }
  }
  testID="card-touchable"
>
  <View
    style={
      {
        "flex": 1,
      }
    }
  />
</View>
`;

exports[`card snapshot card custow arrow color 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "backgroundColor": "#FFFFFF",
      "borderRadius": 12,
      "elevation": 7,
      "flexDirection": "row",
      "minHeight": 128,
      "opacity": 1,
      "padding": 20,
      "shadowColor": "#000",
      "shadowOffset": {
        "height": 3,
        "width": 0,
      },
      "shadowOpacity": 0.29,
      "shadowRadius": 4.65,
    }
  }
  testID="card-touchable"
>
  <View
    style={
      {
        "flex": 1,
      }
    }
  />
  <View
    style={
      {
        "bottom": 0,
        "justifyContent": "center",
        "position": "absolute",
        "right": 20,
        "top": 0,
      }
    }
  >
    <RNSVGSvgView
      align="xMidYMid"
      bbHeight="16"
      bbWidth="16"
      fill="none"
      focusable={false}
      height={16}
      meetOrSlice={0}
      minX={0}
      minY={0}
      style={
        [
          {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          {
            "flex": 0,
            "height": 16,
            "width": 16,
          },
        ]
      }
      vbHeight={16}
      vbWidth={16}
      width={16}
    >
      <RNSVGGroup
        fill={null}
        propList={
          [
            "fill",
          ]
        }
      >
        <RNSVGPath
          d="m10 12 4-4-4-4"
          fill={
            {
              "payload": 4278190080,
              "type": 0,
            }
          }
          propList={
            [
              "stroke",
              "strokeWidth",
              "strokeLinecap",
              "strokeLinejoin",
            ]
          }
          stroke={
            {
              "payload": 4294901760,
              "type": 0,
            }
          }
          strokeLinecap={1}
          strokeLinejoin={1}
          strokeWidth="2"
        />
      </RNSVGGroup>
    </RNSVGSvgView>
  </View>
</View>
`;

exports[`card snapshot card disabled 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": true,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "backgroundColor": "#FFFFFF",
      "borderRadius": 12,
      "elevation": 7,
      "flexDirection": "row",
      "minHeight": 128,
      "opacity": 1,
      "padding": 20,
      "shadowColor": "#000",
      "shadowOffset": {
        "height": 3,
        "width": 0,
      },
      "shadowOpacity": 0.29,
      "shadowRadius": 4.65,
    }
  }
  testID="card-touchable"
>
  <View
    style={
      {
        "flex": 1,
      }
    }
  />
</View>
`;

exports[`card snapshot card no shadow 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "backgroundColor": "#FFFFFF",
      "borderColor": "#F0F0F2",
      "borderRadius": 12,
      "borderWidth": 1,
      "flexDirection": "row",
      "minHeight": 128,
      "opacity": 1,
      "padding": 20,
    }
  }
  testID="card-touchable"
>
  <View
    style={
      {
        "flex": 1,
      }
    }
  />
</View>
`;

exports[`card snapshot card small 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "backgroundColor": "#FFFFFF",
      "borderRadius": 12,
      "elevation": 7,
      "flexDirection": "row",
      "height": 120,
      "opacity": 1,
      "padding": 16,
      "shadowColor": "#000",
      "shadowOffset": {
        "height": 3,
        "width": 0,
      },
      "shadowOpacity": 0.29,
      "shadowRadius": 4.65,
      "width": 135,
    }
  }
  testID="card-touchable"
>
  <View
    style={
      {
        "flex": 1,
      }
    }
  />
</View>
`;

exports[`card snapshot card type highlighted50 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "backgroundColor": "#FFFFFF",
      "borderColor": "#0F4199",
      "borderRadius": 12,
      "borderWidth": 1,
      "elevation": 7,
      "flexDirection": "row",
      "minHeight": 128,
      "opacity": 1,
      "padding": 20,
      "shadowColor": "#000",
      "shadowOffset": {
        "height": 3,
        "width": 0,
      },
      "shadowOpacity": 0.29,
      "shadowRadius": 4.65,
    }
  }
  testID="card-touchable"
>
  <View
    style={
      {
        "flex": 1,
      }
    }
  />
</View>
`;

exports[`card snapshot card type highlighted100 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "backgroundColor": "#013188",
      "borderColor": "#2453A9",
      "borderRadius": 12,
      "borderWidth": 1,
      "elevation": 7,
      "flexDirection": "row",
      "minHeight": 128,
      "opacity": 1,
      "padding": 20,
      "shadowColor": "#000",
      "shadowOffset": {
        "height": 3,
        "width": 0,
      },
      "shadowOpacity": 0.29,
      "shadowRadius": 4.65,
    }
  }
  testID="card-touchable"
>
  <View
    style={
      {
        "flex": 1,
      }
    }
  />
</View>
`;

exports[`card snapshot card with custom style 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "backgroundColor": "#FFFFFF",
      "borderColor": "#0F4199",
      "borderRadius": 12,
      "borderWidth": 1,
      "elevation": 7,
      "flexDirection": "row",
      "minHeight": 128,
      "opacity": 1,
      "padding": 20,
      "shadowColor": "#000",
      "shadowOffset": {
        "height": 3,
        "width": 0,
      },
      "shadowOpacity": 0.29,
      "shadowRadius": 4.65,
    }
  }
  testID="card-touchable"
>
  <View
    style={
      {
        "flex": 1,
      }
    }
  />
</View>
`;

exports[`card snapshot default 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "backgroundColor": "#FFFFFF",
      "borderRadius": 12,
      "elevation": 7,
      "flexDirection": "row",
      "minHeight": 128,
      "opacity": 1,
      "padding": 20,
      "shadowColor": "#000",
      "shadowOffset": {
        "height": 3,
        "width": 0,
      },
      "shadowOpacity": 0.29,
      "shadowRadius": 4.65,
    }
  }
  testID="card-touchable"
>
  <View
    style={
      {
        "flex": 1,
      }
    }
  />
</View>
`;
